### PR TITLE
ADD basic shell:

### PR DESCRIPTION
**Pull request by riblanc**
Adding a basic shell with quote-aware tokenization for improved command parsing.
Includes builtins and help functionality.

### CODE REVIEW

**Comment by riblanc** [comment]
Sorry, the Tokenizer is a quite shady, it'll definitely need a rework later!

**Comment by riblanc** [comment]
Sorry, the tokenizer is a bit shady, it'll definitely need a rework later!

File: src/shell.zig [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		utils.ensure_newline();
+
+		// Search for a builtin command by iterating over those defined in shell/builtins.zig
+    	inline for (@typeInfo(Builtins).Struct.decls) |decl| {
+			if (ft.mem.eql(u8, decl.name, args.tokens[0] orelse "")) {
+				status_code = @field(Builtins, decl.name)(args.tokens[0..args.len]);
+				if (status_code == 0) {
+					continue :main_loop;
+				} else
+					break ; // break the for loop, the builtin command was found but failed
```
Comment (jgiron42) [inline src/shell.zig:43]:
```suggestion
					break ; // break the for loop, the builtin command was found
```

File: src/shell/token.zig [hunk 2]
```diff
@@ -0,0 +1,79 @@
+const ft = @import("../ft/ft.zig");
+const tty = @import("../tty/tty.zig");
+
+const max_tokens = 256;
+
+pub const Tokens = struct {
+	tokens: [max_tokens:null]?[:0]u8 = undefined,
+	len: usize = 0,
+};
```
Comment (jgiron42) [inline src/shell/token.zig:9]: I think you could just define Tokens as a slice of a global array, there is no need for it to be null terminated and len is redundant

**Review by jgiron42 — CHANGES_REQUESTED** [review]
Nice job!

File: src/shell.zig [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+pub fn shell() u8 {
+    tty.get_tty().config.c_lflag.ECHOCTL = true;
+
+	main_loop: while (true) {
+		utils.print_prompt(status_code);
+		status_code = 0;
+
+		// Read a line from the tty
+    	var data: [max_line_size - 1]u8 = undefined;
```
[the remark this answers is not in the document]
Comment (riblanc) [reply src/shell.zig:18]: :thinking: 
:shrug::shrug:

File: src/shell/token.zig (hunk 2 above)
Comment (riblanc) [reply src/shell/token.zig]: Done in 9e00b5d

**Review by jgiron42 — APPROVED** [review]

**Pull request merged**